pkg/liqoctl/install/k3s/provider.go

```diff
@@ -0,0 +1,192 @@
+	k.podCIDR, err = installutils.CheckStringFlagIsSet(flags, providerPrefix, "pod-cidr")
+	if err != nil {
+		return err
+	}
+	klog.V(3).Infof("K3S PodCIDR: %v", k.podCIDR)
+
+	k.serviceCIDR, err = installutils.CheckStringFlagIsSet(flags, providerPrefix, "service-cidr")
+	if err != nil {
+		return err
+	}
+	klog.V(3).Infof("K3S ServiceCIDR: %v", k.serviceCIDR)
+
+	k.apiServer, err = flags.GetString(installutils.PrefixedName(providerPrefix, "api-server"))
+	if err != nil {
+		return err
+	}
+	if k.apiServer != "" {
+		klog.V(3).Infof("K3S API Server: %v", k.apiServer)
+	}
+
+	return nil
+}
+
+// ExtractChartParameters fetches the parameters used to customize the Liqo installation on a specific cluster of a
+// given provider.
+func (k *k3sProvider) ExtractChartParameters(ctx context.Context, config *rest.Config) error {
+	k8sClient, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		fmt.Printf("Unable to create client: %s", err)
+		return err
+	}
+
+	k.k8sClient = k8sClient
+	k.config = config
+
+	if k.apiServer == "" {
+		k.apiServer = config.Host
+	}
+
+	if err = k.validateAPIServer(); err != nil {
+		return err
+	}
+
+	if err = k.validatePodCIDR(ctx); err != nil {
+		return err
+	}
+
+	if err = k.validateServiceCIDR(ctx); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// UpdateChartValues patches the values map with the values required for the selected cluster.
+func (k *k3sProvider) UpdateChartValues(values map[string]interface{}) {
+	values["apiServer"] = map[string]interface{}{
+		"address": k.apiServer,
+	}
+	values["networkManager"] = map[string]interface{}{
+		"config": map[string]interface{}{
+			"serviceCIDR": k.serviceCIDR,
+			"podCIDR":     k.podCIDR,
+		},
+	}
+}
+
+// GenerateFlags generates the set of specific subpath and flags are accepted for a specific provider.
+func GenerateFlags(flags *flag.FlagSet) {
+	subFlag := flag.NewFlagSet(providerPrefix, flag.ExitOnError)
+	subFlag.SetNormalizeFunc(func(f *flag.FlagSet, name string) flag.NormalizedName {
+		return flag.NormalizedName(installutils.PrefixedName(providerPrefix, name))
+	})
+
+	subFlag.String("pod-cidr", defaultPodCIDR, "The Pod CIDR for your cluster (optional)")
+	subFlag.String("service-cidr", defaultServiceCIDR, "The Service CIDR for your cluster (optional)")
+	subFlag.String("api-server", "", "Your cluster API Server URL (optional)")
+
+	flags.AddFlagSet(subFlag)
+}
+
+func (k *k3sProvider) validateServiceCIDR(ctx context.Context) error {
+	svcList, err := k.k8sClient.CoreV1().Services(v1.NamespaceAll).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	_, svcNet, err := net.ParseCIDR(k.serviceCIDR)
+	if err != nil {
+		return err
+	}
+
+	for i := range svcList.Items {
+		svc := &svcList.Items[i]
+		clusterIP := svc.Spec.ClusterIP
+		if clusterIP != "None" && clusterIP != "" && !svcNet.Contains(net.ParseIP(clusterIP)) {
+			return fmt.Errorf("the CIDR %v does not contain IP %v", k.serviceCIDR, clusterIP)
```

Comment:
Would smooth a bit this error, since it will be read by the users. Something along the lines of "It seems the specified service CIDR is not correct as it does not match the services in your cluster", with an higher verbosity message about the actual reason (name of service and IP). Just a proposal though (the same for the pod CIDR)